tests/test_graphnet_backend.py

```diff
@@ -0,0 +1,59 @@
+import pytest
```

Comment:
This test file seems almost the same as the one in `tests/end_to_end/test_end_to_end_graphnet.py`.  I think what you had here before was a subset of the type of tests in `tests/test_tpp_backend.py`, which test just the data extraction from the database using minimal `Cohort` definitions.  

We want to be sure that the data in the graphnet database can be extracted properly, so we want to set up test data that's in the same format as we expect in the graphnet database, and check that the cohort-extractor can retrieve it.  E.g. I know that TPP never has null dates, so if there's a patient in the `practice_registrations` table with no end date, it'll be entered as "9999-12-31", however Graphnet would have a null value there, so we want to test something like:

```
@pytest.mark.integration
def test_registration_dates(database, setup_backend_database):
    setup_backend_database(
        Patient(Patient_ID=1),
        RegistrationHistory(Patient_ID=1, StartDate="2001-01-01", EndDate=None),
        backend="graphnet"
    )

    class Cohort:
        _registrations = table("practice_registrations").first_by("patient_id")
        arrived = _registrations.get("date_start")
        left = _registrations.get("date_end")

    assert extract(Cohort, GraphnetBackend, database) == [
        dict(patient_id=1, arrived=datetime(2001, 1, 1), left=None)
    ]
```

and also (since I think the `date_in_range` method is likely to break with null end dates):
```
@pytest.mark.integration
def test_registration_dates(database, setup_backend_database):
    setup_backend_database(
        Patient(Patient_ID=1),
        RegistrationHistory(Patient_ID=1, StartDate="2001-01-01", EndDate=None),
        Patient(Patient_ID=2),
        RegistrationHistory(Patient_ID=2, StartDate="2001-01-01", EndDate="2020-01-01),
        backend="graphnet"
    )

    class Cohort:
        _registrations = table("practice_registrations").date_in_range("2021-01-01")
       population = _registrations.exists()

    assert extract(Cohort, GraphnetBackend, database) == [
        dict(patient_id=1)
    ]
```

Reply:
Just looked through this now - think there must have been some copy and paste problems happening somewhere as I definitely had a different set of tests. Let me look into that...